instant-search/react-instantsearch/src/App.js

```diff
@@ -28,133 +27,172 @@ const App = props =>
+        />
+        <h5>Price</h5>
+        <ConnectedRange attributeName="price" />
+        <h5>Type</h5>
+        <Menu attributeName="type" />
+      </div>
+      {rightColumn}
     </div>
-    {rightColumn}
-  </div>);
+  );
 });
 
 class Range extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {currentValues: {min: this.props.min, max: this.props.max}};
+    this.state = {
+      currentValues: { min: this.props.min, max: this.props.max },
+    };
     this.onValuesUpdated = this.onValuesUpdated.bind(this);
     this.onChange = this.onChange.bind(this);
   }
+
   componentWillReceiveProps(sliderState) {
-    this.setState({currentValues: {min: sliderState.currentRefinement.min, max: sliderState.currentRefinement.max}});
+    this.setState({
+      currentValues: {
+        min: sliderState.currentRefinement.min,
+        max: sliderState.currentRefinement.max,
+      },
+    });
   }
+
   onValuesUpdated(values) {
-    this.setState({currentValues: {min: values[0], max: values[1]}});
+    this.setState({ currentValues: { min: values[0], max: values[1] } });
   }
+
   onChange(values) {
-    if (this.props.currentRefinement.min !== values[0] || this.props.currentRefinement.max !== values[1]) {
-      this.props.refine({min: values[0], max: values[1]});
+    if (
+      this.props.currentRefinement.min !== values[0] ||
+      this.props.currentRefinement.max !== values[1]
+    ) {
+      this.props.refine({ min: values[0], max: values[1] });
     }
   }
+
   render() {
-    const {min, max} = this.props;
-    const {currentValues} = this.state;
+    const { min = 0, max = 0 } = this.props;
+    const { currentValues } = this.state;
+
     return (
       <div className="ais-Slider__root">
-        <Slider
-          max={Math.trunc(max)} min={Math.trunc(min)}
-          onAfterChange={this.onChange}
-          onChange={this.onValuesUpdated}
-          range
-          value={[currentValues.min, currentValues.max]}
-        />
+        {currentValues.min === undefined && currentValues.max === undefined
+          ? null
+          : <Slider
+              max={Math.trunc(max)}
+              min={Math.trunc(min)}
+              onAfterChange={this.onChange}
+              onChange={this.onValuesUpdated}
+              range
+              value={[currentValues.min, currentValues.max]}
+            />}
         <div className="ais-Slider__values">
-          <div>{min}</div>
-          <div>{max}</div>
+          <div>
+            {min}
+          </div>
+          <div>
+            {max}
+          </div>
         </div>
       </div>
     );
   }
 }
 
 Range.propTypes = {
-  currentRefinement: React.PropTypes.object.isRequired,
-  max: React.PropTypes.number.isRequired,
-  min: React.PropTypes.number.isRequired,
-  refine: React.PropTypes.func.isRequired
+  currentRefinement: PropTypes.object,
+  max: PropTypes.number,
+  min: PropTypes.number,
+  refine: PropTypes.func,
```

Comment:
I had to change these PropTypes because it was undefined at first render. What is the value in the storybooks @mthuret